graceful_controller_ros/src/graceful_controller_ros.cpp

```diff
@@ -359,17 +360,26 @@ class GracefulControllerROS : public nav_core::BaseLocalPlanner
       tf2::doTransform(transformed_plan[i], target_pose, odom_to_base);
 
       // Continue if target_pose is too far away from robot
-      if (std::hypot(target_pose.pose.position.x, target_pose.pose.position.y) > max_lookahead_)
+      double dist_to_target = std::hypot(target_pose.pose.position.x, target_pose.pose.position.y);
+      if (dist_to_target > max_lookahead_)
       {
         continue;
       }
 
-      // Avoid unstability and big sweeping turns at the end of paths by ignoring final heading
-      if (prefer_final_rotation_ && (dist_to_goal < max_lookahead_))
+      if (dist_to_goal < max_lookahead_)
       {
-        double yaw = std::atan2(target_pose.pose.position.y, target_pose.pose.position.x);
-        target_pose.pose.orientation.z = sin(yaw / 2.0);
-        target_pose.pose.orientation.w = cos(yaw / 2.0);
+        if (prefer_final_rotation_)
+        {
+          // Avoid unstability and big sweeping turns at the end of paths by ignoring final heading
+          double yaw = std::atan2(target_pose.pose.position.y, target_pose.pose.position.x);
+          target_pose.pose.orientation.z = sin(yaw / 2.0);
+          target_pose.pose.orientation.w = cos(yaw / 2.0);
+        }
+      }
+      else if (dist_to_target < min_lookahead_)
```

Comment:
No, if the dist_to_goal < max_likelihood, then we use the orientation of the goal, rather than the angle between current pose and goal, so the risk of rotating is gone. Also, this would abort before we ever get to the goal (we'd only be able to get to min_lookahead_ distance away from the goal)

Reply:
We don't want to abort when dist_to_target < min_lookahead_ because we are at the end of the path (otherwise min_lookahead_ becomes a new, possibly higher goal tolerance).

Reply:
yeah that makes sense